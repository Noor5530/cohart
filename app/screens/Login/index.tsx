import { useNavigation, useRoute } from "@react-navigation/native";
import Arrow from "assets/Arrow_Left.png";
import StatusBar from "components/CustomStatusBar";
import TextField from "components/PhoneTextInput";
import { RequestOTPType } from "enums/request_otp_type";
import { IRequestOtpPayload } from "models/actions/user";
import { LoginScreenRouteProps } from "navigation/types";
import React from "react";
import {
  Image,
  ImageBackground,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view";
import PhoneInput from "react-native-phone-number-input";
import AntDesign from "react-native-vector-icons/AntDesign";
import { useDispatch, useSelector } from "react-redux";
import { requestOtp } from "store/actions/userActions";

import { textColor } from "../../config/colors";
import AppState from "../../models/reducers";
import styles from "./styles";

const Login: React.FC = () => {
  const navigation = useNavigation();
  const route = useRoute<LoginScreenRouteProps>();
  const [isTermsConditionSelected, setTermsCondition] =
    React.useState<boolean>(false);
  const token = useSelector((state: AppState) => state.user.token);
  const [phoneNumber, setPhoneNumber] = React.useState<string>("");
  const [error, setError] = React.useState<string>("");
  const dispatch = useDispatch();
  const phoneInput = React.useRef<PhoneInput>(null);
  const [formattedValue, setFormattedValue] = React.useState("");
  const onSubmit = () => {
    const regex = /^\+(?:[0-9] ?){6,14}[0-9]$/;

    if (regex.test(formattedValue)) {
      const requestOtpType = route?.params?.data?.user_code
        ? RequestOTPType.UserCode
        : RequestOTPType.PhoneNumber;
      const payload: IRequestOtpPayload = {
        request_otp_type: requestOtpType,
        user_code: route?.params?.data?.user_code,
        phone_number: formattedValue,
        non_formatted_phone_number: phoneNumber,
        country_code: formattedValue.slice(0, -phoneNumber.length),
        registration_id: token,
      };
      dispatch(requestOtp(payload));
    } else {
      setError("Please enter valid number");
    }
  };
  const goBack = () => navigation.goBack();

  return (
    <KeyboardAwareScrollView key="login" style={{ flex: 1 }}>
      <StatusBar backgroundColor={textColor} barStyle="dark-content" />

      <ImageBackground
        style={styles.imageBackground}
        source={require("../../assets/LoginBackGround.png")}
      >
        <View style={styles.container}>
          <Text style={styles.shortText}>
            {route?.params?.code?.length > 0
              ? "Please enter your phone number to sign in to the community"
              : "Please enter your phone number to sign in to the community"}
          </Text>
          <TextField
            defaultCode={"US"}
            phoneInput={phoneInput}
            onChangeFormattedText={(text) => {
              setFormattedValue(text);
            }}
            isNumber={true}
            textContentType="telephoneNumber"
            dataDetectorTypes="phoneNumber"
            keyboardType="phone-pad"
            placeholderTextColor={"rgba(0,0,0,0.2)"}
            onSelectCountry={(text) => {
              console.log("onSelectCountry", text);
            }}
            maxLength={12}
            defaultValue={phoneNumber}
            setValue={setPhoneNumber}
            isValidate={isTermsConditionSelected}
            setIsValidate={setTermsCondition}
            onSubmit={onSubmit}
            placeholder={"xxxxxxxxxx"}
            error={error}
            setError={setError}
            onChangeText={(text) => {
              setError("");
              setPhoneNumber(text);
            }}
          />

          <View style={styles.checkContainer}>
            <TouchableOpacity
              style={styles.checkBox}
              onPress={() => {
                setTermsCondition((prv) => !prv);
              }}
            >
              {isTermsConditionSelected && (
                <AntDesign name={"check"} color="black" size={15} />
              )}
            </TouchableOpacity>
            <Text
              style={styles.footer}
              onPress={() => {
                navigation.navigate("TermsAndCondition");
              }}
            >
              {"I accept the  "}
              <Text
                style={styles.termsConditionLine}
                onPress={() => {
                  navigation.navigate("TermsAndCondition");
                }}
              >
                {"Terms & Conditions"}
              </Text>
            </Text>
          </View>

          <Text
            style={styles.referral}
            onPress={() => {
              navigation.navigate("SignUp");
            }}
          >
            [ No Invite? ]
          </Text>
          <TouchableOpacity onPress={goBack} style={{ marginTop: 40 }}>
            <Image source={Arrow} />
          </TouchableOpacity>
          <Text style={styles.bottomText}>
            Featured artist:
            <Text style={styles.bottomInnerText}> alec demarco </Text>
          </Text>
        </View>
      </ImageBackground>
    </KeyboardAwareScrollView>
  );
};

export default Login;
