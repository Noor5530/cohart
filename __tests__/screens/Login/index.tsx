import 'react-native';
import React from 'react'; 
import Login from '../../../app/screens/Login';
import renderer from 'react-test-renderer';

jest.useFakeTimers()

test('renders correctly', () => {

    const tree = renderer.create(<Login />).toJSON();

    expect(tree).toMatchSnapshot();

});
