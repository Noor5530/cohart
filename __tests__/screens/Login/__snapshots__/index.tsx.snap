// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/assets/LoginBackGround.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "paddingTop": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "flex-end",
          "color": "#FFFFFF",
          "fontFamily": "Inter-Regular",
          "fontSize": 18.0084,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": 22,
          "textAlign": "center",
          "width": 298.94,
        }
      }
    >
      Please enter your phone number to sign in to the community
    </Text>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingTop": 22.81,
            "width": "100%",
          },
          Object {
            "paddingLeft": undefined,
            "paddingRight": undefined,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "flexDirection": "row",
              "width": 600,
            },
            Object {},
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#F6F6F6",
              "borderColor": " #000000",
              "borderRadius": 14.4067,
              "height": 50.42,
              "justifyContent": "flex-start",
              "margin": 0,
              "padding": 0,
              "paddingLeft": 20,
              "width": "100%",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F6F6F6",
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <mockedDefaultExport
            color="black"
            name="down"
            size={20}
          />
          <mockedDefaultExport
            countryCode="US"
            filterProps={Object {}}
            onClose={[Function]}
            onSelect={[Function]}
            renderFlagButton={[Function]}
            visible={false}
            withCallingCode={true}
            withEmoji={true}
            withFilter={true}
            withFlag={true}
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Inter-Regular",
                  "fontSize": undefined,
                  "margin": 0,
                  "paddingRight": 5,
                },
              ]
            }
          >
            +undefined
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F8F9F9",
                "flex": 1,
                "flexDirection": "row",
                "paddingHorizontal": 30,
                "paddingVertical": 27,
                "textAlign": "left",
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "borderBottomEndRadius": 14.4067,
                "borderTopRightRadius": 14.4067,
                "height": 50.42,
                "margin": 0,
                "width": "90%",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardAppearance="default"
            keyboardType="number-pad"
            maxLength={12}
            onChangeText={[Function]}
            placeholder="xxxxxxxxxx"
            rejectResponderTermination={true}
            selectionColor="black"
            style={
              Array [
                Object {
                  "color": "#000000",
                  "flex": 1,
                  "fontSize": 16,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "Inter-Regular",
                  "fontSize": undefined,
                  "height": 50.42,
                  "margin": 0,
                  "padding": 0,
                  "width": "90%",
                },
              ]
            }
            testID="phoneInputText"
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingTop": 23,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#000000",
            "borderRadius": 5.7147,
            "borderWidth": 0.476225,
            "height": 20,
            "justifyContent": "center",
            "opacity": 1,
            "width": 20,
          }
        }
      />
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": 17,
            "paddingLeft": 10,
            "textAlign": "center",
          }
        }
      >
        I accept the  
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "#E6FF00",
              "fontFamily": "Inter-Regular",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": 17,
              "paddingLeft": 3,
              "textAlign": "center",
              "textDecorationColor": "#E6FF00",
              "textDecorationLine": "underline",
            }
          }
        >
          Terms & Conditions
        </Text>
      </Text>
    </View>
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "#E6FF00",
          "fontFamily": "Inter-Regular",
          "fontSize": 15.6501,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "height": 23,
          "lineHeight": 23,
          "marginTop": undefined,
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      [ No Invite? ]
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginTop": 40,
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/Arrow_Left.png",
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "bottom": undefined,
          "color": "#FFFFFF",
          "fontFamily": "Inter-Regular",
          "fontSize": 10.955,
          "fontStyle": "normal",
          "fontWeight": "400",
          "left": 41.24,
          "lineHeight": 16,
          "position": "absolute",
          "textTransform": "uppercase",
        }
      }
    >
      Featured artist:
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Inter-Regular",
            "fontSize": 10.955,
            "fontStyle": "normal",
            "fontWeight": "600",
            "lineHeight": 16,
            "textTransform": "uppercase",
          }
        }
      >
         alec demarco 
      </Text>
    </Text>
  </View>
</View>
`;
